channels/sig_analog.c

```diff
@@ -1087,12 +1089,29 @@ int analog_call(struct analog_pvt *p, struct ast_channel *ast, const char *rdest
 		}
 
 		if (p->use_callerid) {
+			const char *qual_var;
+
+			/* Caller ID Name and Number */
 			p->caller.id.name.str = p->lastcid_name;
 			p->caller.id.number.str = p->lastcid_num;
 			p->caller.id.name.valid = ast_channel_connected(ast)->id.name.valid;
 			p->caller.id.number.valid = ast_channel_connected(ast)->id.number.valid;
 			p->caller.id.name.presentation = ast_channel_connected(ast)->id.name.presentation;
 			p->caller.id.number.presentation = ast_channel_connected(ast)->id.number.presentation;
+
+			/* Redirecting Reason */
+			p->redirecting_reason = ast_channel_redirecting(ast)->from.number.valid ? ast_channel_redirecting(ast)->reason.code : -1;
+
+			/* Call Qualifier */
+			ast_channel_lock(ast);
+			/* XXX In the future, we may want to make this a CALLERID or CHANNEL property and fetch it from there. */
+			qual_var = pbx_builtin_getvar_helper(ast, "CALL_QUALIFIER");
```

Comment:
This is not documented anywhere.

Reply:
What would be the best way to document it? It's not part of an application, so there's not really an appropriate home I can think of for it in the xmldocs. It isn't a CHANNEL function variable, it's not even chan_dahdi specific, per se.

It is Caller ID related, however. Would it be appropriate to document it in `func_callerid`, even though it's not currently set there? Might also be useful if it's eventually migrated to be part of the caller ID structure.

Reply:
It is currently only applicable in this case, so it is chan_dahdi specific therefore honestly probably multiple places. The DAHDI configuration file and func_callerid with explicit statement that it only works on DAHDI.